app/src/components/common/badges/Badge.tsx

```diff
@@ -0,0 +1,45 @@
+import { Badge as BaseBadge } from "@/components/ui/badge"
+import { Check, Loader2, XIcon } from "lucide-react"
+import { ReactNode } from "react"
+
+const icons = {
+  loader: <Loader2 width={12} height={12} />,
+  check: <Check width={12} height={12} />,
+  X: <XIcon width={12} height={12} />,
+}
+
+const colorSchemes = {
+  blue: {
+    background: "bg-calloutAlternative-foreground",
+    text: "text-blue-800",
+  },
+  green: {
+    background: "bg-green-100",
+    text: "text-green-800",
+  },
+  red: {
+    background: "bg-rose-200",
+    text: "text-rose-800",
+  },
+}
+export const Badge = ({
+  content = "",
+  icon,
+  colorScheme,
+}: {
+  icon?: keyof typeof icons
+  content: ReactNode
+  colorScheme?: keyof typeof colorSchemes
+}) => {
+  return (
+    <BaseBadge
+      className={`text-xs font-medium border-0 flex gap-1 ${
```

Comment:
I would suggest using `cn` instead of formatting a string like this. It improves maintainability and handles faulty values that are passed in, in case it happens
```suggestion
      className={cn(
        "text-xs font-medium border-0 flex gap-1",
        colorScheme && colorSchemes[colorScheme]?.background,
        colorScheme && colorSchemes[colorScheme]?.text
      )}
```

Reply:
I'd like to expand on this conversation a bit.

Typically I've written component classes as such that you can wrap other classes around the components if you needed something more custom.

For instance:


```
export function MyComponent() {
    return <div className="w-full"/>
}

...somewhere else:
<div className="w-[400px]">
    <MyComponent />
</div>
```

This approach doesn't require the component to be set up with the customized className parameter and associated `cn()` function.

Additionally, this approach allows for a better developer experience to the implementer of the component by providing human-friendly parameters (i.e they are passing in `blue` or `red` instead of something like: `bg-calloutAlternative-foreground text-blue-800`.

Is there a benefit that I'm not seeing with the `cn()` approach or a drawbacks in my approach?

Reply:
I am not sure I follow how does suggested implementation obstruct us from having responsive components? 

Benefit to (and mostly the point of) using `cn` is to merge multiple style strings and/or conditionally render styles. In my opinion approach to color scheme here is wrong and variants should be extracted in a separate `Record` type object where we have `default` and all the otheres - then `colorScheme` would default to `default` value and hence we wouldnt have to conditionally check for it like we do here:

```
colorScheme && colorSchemes[colorScheme]?.background,
colorScheme && colorSchemes[colorScheme]?.text
```

You can check something similar here: https://github.com/voteagora/op-atlas/blob/a02f1e2403641d83397dcaac1df9fdc14f28151d/app/src/components/common/Button.tsx#L32

This component's styling is kind of readable at this point with a formatted string, but extending it further, especially if there are more conditionals in play is going to be hard to read, which `cn` can handle this for us already.

Maintainability is a big thing to consider when developing abstracted, customisable components.